tensordict/nn/common.py

```diff
@@ -923,17 +941,35 @@ def __init__(
         in_keys: NestedKey | List[NestedKey] | Dict[NestedKey:str],
         out_keys: NestedKey | List[NestedKey],
         *,
+        out_to_in_map: bool | None = None,
         inplace: bool | str = True,
     ) -> None:
         super().__init__()
 
+        if out_to_in_map is not None and not isinstance(in_keys, dict):
+            warnings.warn(
+                "out_to_in_map is not None but is only used when in_key` is a dictionary."
+            )
+
         if isinstance(in_keys, dict):
+            if out_to_in_map is None:
+                warnings.warn(
+                    "Using a dictionary in_keys without specifying out_to_in_map is deprecated."
+                    "Use out_to_in_map to indicate the ordering of the input keys.",
```

Comment:
I think we want to change the behavior - currently `out_to_in_map` is `False` by default but we would like to make it `True` eventually. This should be mentioned in the warning, as well as the version where the change will occur (v0.9).